src/train.py

```diff
@@ -0,0 +1,356 @@
+    font: str = "resources/fonts/NotoSans.ttf",
+    out_channels: int = 256,
+    nhead: int = 12,
+    beta1: float = 0.9,
+    beta2: float = 0.999,
+    sl: bool = False,
+    dropout: float = 0.0,
+    device: Optional[str] = None,
+    warmup: int = 1000,
+    batch_size: int = 32,
+    kernel_size: int = 3,
+    num_layers: int = 1,
+    lr: float = 5e-5,
+):
+    """
+    Trains model
+
+    :param train_data: path to training data (default: resources/data/train_dataset.jsonl)
+    :param val_data: path to validation data (default: resources/data/val_dataset.jsonl)
+    :param test_data: path to test data (default: resources/data/test_dataset.jsonl)
+    :param tokenizer: path to pretrained tokenizer directory. If None, vtr model will be trained (default: None)
+    :param max_seq_len: used to create datasets. In the case of vtr, a limit on the number of slices, in the case of a classic transformer, on the number of tokens. (default: 512)
+    :param max_slices_count_per_word: used to limit the number of slices for one word. You can learn more about the process of processing a sequence of words by looking at the image `resources/images/padding_in_sl.jpg` (default: 9)
+    :param font_size: font size with which the image is generated (default: 15)
+    :param window_size: window width (default: 30)
+    :param stride: window step size (default: 5)
+    :param epochs: number of training epochs (default: 10)
+    :param random_state: (default: 21)
+    :param log_every: how often to recalculate the value of metrics on the validation set. These values are logged to wandb (default: 1000)
+    :param emb_size: embedding size received by visual representations (default: 768)
+    :param num_workers: number of threads in DataLoader (default: 1)
+    :param save_to: the path where the weights of the model are saved after training, if a value is specified, if not, then the weights are not saved (default: None)
+    :param font: font used to generate character images (default: resources/fonts/NotoSans.ttf)
+    :param out_channels: number of output channels in the last convolutional layer (default: 256)
+    :param nhead: number of transformer layer heads (default: 12)
+    :param beta1: Adam optimizer parameter (default: 0.9)
+    :param beta2: Adam optimizer parameter (default: 0.999)
+    :param sl: if true, then the task of labeling the sequence is being solved (default: False)
+    :param dropout: dropout probability (default: 0.0)
+    :param device: 'cuda' or 'cpu'. If None it will be set automatically (default: None)
+    :param warmup: number of steps to warm up Linear Scheduler (default: 1000)
+    :param batch_size: batch size (default: 32)
+    :param kernel_size: convolution kernel size (default 3)
+    :param num_layers: number of transformer encoder layers in the visual representation model (default: 1)
+    :param lr: learning rate (default: 5e-5)
+    :return:
+    """
+    wandb.init(project="visual-text", entity="borisshapa")
```

Comment:
`entity="borisshapa"` -- this will not work for other users. Btw, this can be omitted since everyone has already logged in to the account.